File: 7.12.6/gwan/0.0.0.0:80/#0.0.0.0/old.csp/hello.cpp
// ============================================================================
// C++ servlet example for the G-WAN Web Application Server   (http://gwan.ch/)
// ----------------------------------------------------------------------------
// hello.cpp: hello world C++ example
//
// This code was submitted and written by Thomas Meitz (gwan at jproxx dot com)
// ============================================================================
// imported functions:
//   get_reply(): get a pointer on the 'reply' dynamic buffer from the server
//   xbuf_cat(): like strcat(), but it works in the specified dynamic buffer
// ----------------------------------------------------------------------------
#include "gwan.h" // G-WAN definitions
#include <string>

using namespace std;

class Hello 
{
public: 
   void whatsup(xbuf_t* reply, string& str)
   {
      xbuf_cat(reply, (char*)str.c_str());
   }
};
// ----------------------------------------------------------------------------
int main(int argc, char *argv[])
{
   string h("Hello World (C++)");
   Hello hello;
   hello.whatsup(get_reply(argv), h);
   
   return 200; // return an HTTP code (200:'OK')
}
// ============================================================================
// End of Source Code
// ============================================================================

File: 7.12.6/gwan/0.0.0.0:80/#0.0.0.0/old.csp/hello.cs
// ============================================================================
// C servlet sample for the G-WAN Web Application Server (http://trustleap.ch/)
// ----------------------------------------------------------------------------
// hello.cs: G-WAN supports .NET C# servlets
// ============================================================================
using System;

public class hello
{
   public static int Main(string[] args)
   {
      Gwan.xbufCat(Gwan.getReply(args[0]), "Hello, C#!");
      return 200; // HTTP status (200:'OK')
   }
}
// ============================================================================
// End of Source Code
// ============================================================================
